### PR TITLE
[chore] deprecate signalfxexporter max_connections warning

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Pendant to #16807, we now have a better way to set up the http client settings, but we need to finish deprecating the max_connections config option. This PR removes use of MaxConnections and adds a warning if it is explicitly used.

**Link to tracking Issue:**
#16807

**Testing:**
Tests pass.

**Documentation:**
No changes.

### CODE REVIEW

File: exporter/signalfxexporter/exporter.go [hunk 1]
```diff
@@ -230,10 +230,9 @@ func (se *signalfxExporter) startLogs(_ context.Context, host component.Host) er
 func (se *signalfxExporter) createClient(host component.Host) (*http.Client, error) {
 	se.config.HTTPClientSettings.TLSSetting = se.config.IngestTLSSettings
 
-	if se.config.HTTPClientSettings.MaxIdleConns == nil {
+	if se.config.MaxConnections != 0 {
+		se.logger.Warn("You are using the deprecated `max_connections` option that will be removed in the next release; use `max_idle_conns` and `max_idle_conns_per_host` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
```
Comment (dmitryax) [inline exporter/signalfxexporter/exporter.go:234]:
```suggestion
		se.logger.Warn("You are using the deprecated `max_connections` option that will be removed soon; use `max_idle_conns` and `max_idle_conns_per_host` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
```

File: exporter/signalfxexporter/exporter.go [hunk 2]
```diff
@@ -230,10 +230,9 @@ func (se *signalfxExporter) startLogs(_ context.Context, host component.Host) er
...
 		se.config.HTTPClientSettings.MaxIdleConns = &se.config.MaxConnections
```
Comment (dmitryax) [inline exporter/signalfxexporter/exporter.go:235]: `MaxIdleConns` should have precedence over the deprecated `MaxConnections`, so we need an additional `se.config.HTTPClientSettings.MaxIdleConns != nil` condition here. Same goes to `MaxIdleConnsPerHost`

Comment (atoulme) [reply exporter/signalfxexporter/exporter.go:235]: Makes sense! On it.

File: exporter/signalfxexporter/exporter.go [hunk 3]
```diff
@@ -230,10 +230,12 @@ func (se *signalfxExporter) startLogs(_ context.Context, host component.Host) er
...
-	if se.config.HTTPClientSettings.MaxIdleConns == nil {
+	if se.config.HTTPClientSettings.MaxIdleConns == nil && se.config.MaxConnections != 0 {
+		se.logger.Warn("You are using the deprecated `max_connections` option that will be removed soon; use `max_idle_conns` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
 		se.config.HTTPClientSettings.MaxIdleConns = &se.config.MaxConnections
 	}
-	if se.config.HTTPClientSettings.MaxIdleConnsPerHost == nil {
+	if se.config.HTTPClientSettings.MaxIdleConnsPerHost == nil && se.config.MaxConnections != 0 {
+		se.logger.Warn("You are using the deprecated `max_connections` option that will be removed soon; use `max_idle_conns_per_host` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
 		se.config.HTTPClientSettings.MaxIdleConnsPerHost = &se.config.MaxConnections
 	}
```
Comment (dmitryax) [inline exporter/signalfxexporter/exporter.go:240]: It'll bring two messages to everyone who uses just `max_connections`, which is the most common use case. I think it should be

```golang
if  se.config.MaxConnections != 0 && (se.config. MaxIdleConns == nil || se.config.HTTPClientSettings.MaxIdleConnsPerHost == nil) {
	se.logger.Warn("You are using the deprecated `max_connections` option that will be removed soon; use `max_idle_conns` and/or `max_idle_conns_per_host` instead: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/signalfxexporter#advanced-configuration")
	if se.config.HTTPClientSettings.MaxIdleConns == nil {
		se.config.HTTPClientSettings.MaxIdleConns = &se.config.MaxConnections
	}
	if se.config.HTTPClientSettings. MaxIdleConnsPerHost == nil {
		se.config.HTTPClientSettings. MaxIdleConnsPerHost = &se.config.MaxConnections
	}
}
```

Comment (atoulme) [reply exporter/signalfxexporter/exporter.go]: Makes sense, applied.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**